networking/external_dns_operator/nw-external-dns-operator-configuration-parameters.adoc

```diff
@@ -0,0 +1,88 @@
+[id="nw-external-dns-operator-configuration-parameters_{context}"]
+= External DNS Operator configuration parameters
+
+[cols="3a,8a",options="header"]
+|===
+|Parameter |Description
+
+|`spec`
+|`spec` parameter enables
+
+[source,yaml]
+----
+spec:
+  provider:
+    type: AWS <1>
+    aws:
+      credentials:
+        name: aws-access-key <2>
+----
+<1> Available options are AWS, GCP and Azure.
+<2> Name of the configmap which has the credentials for your cloud provider.
+
+|`zones`
+|`zones` enables your specified AWS zone. When you do not specify zones, ExternalDNS discovers all the zones in the account.
+
+[source,yaml]
+----
+zones:
+- "myzoneid" <1>
+----
+
+<1> Specify the ids of hosted zones where you want to publish the dns records for your source i.e service or route.
+
+|`domains`
+|`domains` parameter tells the external dns which hostnames with a particular domain shall be considered for record creation to the zone specified in zones section.
+
+[source,yaml]
+----
+domains:
+- filterType: Include <1>
+  matchType: Exact <2>
+  name: "myzonedomain1.com" <3>
+- filterType: Include
+  matchType: Pattern <4>
+  pattern: ".*\\.otherzonedomain.com" <5>
+----
+<1> Include checks if the domain is included as a part of the hostname.
+<2> If the source is route then the route hostname's domain will be checked for the name defined in the name field. So, as per this example if an hostname is test.myzonedomain1.com or apps.test.myzonedomain1.com then it will be filtered by external dns for dns record creation in the defined zone ID's present under zone's section.
+<3> Define the domain name to be filtered.
+<4> Sets "regex-domain-filter" in external-dns. Limit possible domains and target zones by a Regex filter.
+<5> Define the regex pattern to be used by external to filter domains and target zones.
+
+|`source`
+|`source`
+
+[source,yaml]
+----
+source: <1>
+  type: Service <2>
+  service:
+    serviceType:<3>
+      - LoadBalancer
+      - ClusterIP
+  labelFilter: <4>
+    matchLabels:
+      external-dns.mydomain.org/publish: "yes"
+  hostnameAnnotation: "Allow" <5>
+  fqdnTemplate:
+  - "{{.Name}}.myzonedomain.com" <6>
+----
+
+<1> The resource types that are queried for endpoints.
+<2> External DNS will filter source type of service for creating dns records.
+<3> It sets "service-type-filter" in external-dns. The service types to take care about (default: all, expected: ClusterIP, NodePort, LoadBalancer or ExternalName)
+<4> Filter sources managed by external-dns via label selector when listing all resources; currently supported by source types CRD, ingress, service and openshift-route".
```

Comment:
The description of `<4>` is incorrect. The `labelFilter` field is used to specify a [LabelSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#labelselector-v1-meta). It can have `matchLabels` or `matchExpression` and this selector is used to filter out resources which don't have the expected labels.